migration/v0.42/prepare/40-azure-sync-destination-type.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+
+HERE="$(dirname "$(readlink -f "${0}")")"
+ROOT="$(readlink -f "${HERE}/../../../")"
+
+# shellcheck source=scripts/migration/lib.sh
+source "${ROOT}/scripts/migration/lib.sh"
+
+log_info "configuring object storage sync type on azure"
+if [[ "${CK8S_CLUSTER}" =~ ^(sc|both)$ ]]; then
+  cloud_provider=$(yq_dig sc .global.ck8sCloudProvider)
+  if [[ "${cloud_provider}" != "azure" ]]; then
+    log_info "not running on azure, skipping"
+    exit 0
+  fi
+
+  if yq_check sc .objectStorage.sync.enabled true; then
+    if yq_null sc .objectStorage.sync.destinationType; then
+      log_info "objectStorage sync is enabled, will keep previous default destinationType: s3"
+      yq_add sc .objectStorage.sync.destinationType "\"s3\""
```

Comment:
If sync is enabled, then it would rather quickly be clear that it is not working if you were intending to sync to azure, since the jobs would constantly fail.
For our managed service it is also likely that the sync will be an of-cloud sync to something that is using s3.
So I think this is actually fine and I will this time be lazy and not add the prompt.